pkgs/build-support/node/build-npm-package/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib, stdenv, fetchNpmDeps, npmHooks, nodejs }:
+
+{ name ? "${args.pname}-${args.version}"
+, src ? null
+, srcs ? null
+, sourceRoot ? null
+, patches ? [ ]
+, nativeBuildInputs ? [ ]
+, buildInputs ? [ ]
+  # The output hash of the dependencies for this project.
+  # Can be calculated in advance with prefetch-npm-deps.
+, npmDepsHash ? ""
+  # Whether to make the cache writable prior to installing dependencies.
+  # Don't set this unless npm tries to write to the cache directory, as it can slow down the build.
+, makeCacheWritable ? false
+  # The script to run to build the project.
+, npmBuildScript ? "build"
+  # Flags to pass to all npm commands.
+, npmFlagsArray ? [ ]
+  # Flags to pass to `npm ci`.
+, npmInstallFlagsArray ? [ ]
+  # Flags to pass to `npm run ${npmBuildScript}`.
+, npmBuildFlagsArray ? [ ]
+  # Flags to pass to `npm pack`.
+, npmPackFlagsArray ? [ ]
```

Comment:
It's good to have this documented here as well. Please update the strings, even if this is double book-keeping for now. The general problem may be a worthwhile for the Nixpkgs architecture team (@infinisil) and Nix Documentation team (@domenkozar) to address.